src/Sharpliner/AzureDevOps/Model/Tasks/Marketplace/NuGetPackCommandTask.cs

```diff
@@ -0,0 +1,298 @@
+
+        /// <summary>
+        /// Normal verbosity.
+        /// </summary>
+        Normal,
+
+        /// <summary>
+        /// Detailed verbosity.
+        /// </summary>
+        Detailed,
+    }
+
+    /// <summary>
+    /// Represents the NuGetCommand@2 task for packing NuGet packages in Azure DevOps pipelines with <c>versioningScheme></c> set to <c>off</c>.
+    /// </summary>
+    public record NuGetPackCommandTaskOff : NuGetPackCommandTask
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NuGetPackCommandTaskOff"/> class.
+        /// </summary>
+        public NuGetPackCommandTaskOff() : base("off") { }
+
+        /// <summary>
+        /// Includes referenced projects either as dependencies or as part of the package. 
+        /// Cannot be used with automatic package versioning. 
+        /// If a referenced project has a corresponding <c>nuspec</c> file that has the same name as the project, then that referenced project is added as a dependency. 
+        /// Otherwise, the referenced project is added as part of the package. 
+        /// Learn more about <see href="https://learn.microsoft.com/en-us/nuget/tools/cli-ref-pack">using the pack command for NuGet CLI to create NuGet packages</see>.
+        /// </summary>
+        [YamlIgnore]
+        public bool IncludeReferencedProjects
+        {
+            get => GetBool("includeReferencedProjects", false);
+            init => SetProperty("includeReferencedProjects", value);
+        } 
+    }
+
+    /// <summary>
+    /// Represents the NuGetCommand@2 task for packing NuGet packages in Azure DevOps pipelines with <c>versioningScheme></c> set to <c>byPrereleaseNumber</c>.
+    /// </summary>
+    public record NuGetPackCommandTaskByPrereleaseNumber : NuGetPackCommandTask
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NuGetPackCommandTaskByPrereleaseNumber"/> class.
+        /// </summary>
+        /// <param name="majorVersion">The <c>X</c> in version <see href="http://semver.org/spec/v1.0.0.html">X.Y.Z</see>.</param>
+        /// <param name="minorVersion">The <c>Y</c> in version <see href="http://semver.org/spec/v1.0.0.html">X.Y.Z</see>.</param>
+        /// <param name="patchVersion">The <c>Z</c> in version <see href="http://semver.org/spec/v1.0.0.html">X.Y.Z</see>.</param>
+        public NuGetPackCommandTaskByPrereleaseNumber(string majorVersion, string minorVersion, string patchVersion) : base("byPrereleaseNumber")
+        {
+            MajorVersion = majorVersion;
+            MinorVersion = minorVersion;
+            PatchVersion = patchVersion;
+        }
+
+        /// <summary>
+        /// The <c>X</c> in version <see href="http://semver.org/spec/v1.0.0.html">X.Y.Z</see>.
+        /// </summary>
+        [YamlIgnore]
+        public string MajorVersion
+        {
+            get => GetString("majorVersion")!;
+            init => SetProperty("majorVersion", value);
+        }
```

Comment:
I think all of these should rather be

```suggestion
        public string? MajorVersion
        {
            get => GetString("majorVersion");
            init => SetProperty("majorVersion", value);
        }
```

Reply:
these version arguments are required will be set through the ctor, are you sure here? I think the value here is never null

Reply:
Ah, I see. For those coming from a ctor we shouldn't then, yeah, or we should maybe even do a nullcheck in the setter?

Reply:
do we have a standard way of requiring task properties to be non-null?

Reply:
I don't think we do but if we have a setter on a required property, it should maybe throw? Because the YAML it will produce will technically be wrong?

Reply:
I think we could throw a ArgumentException